src/frontend/hooks/goose-form.js

```diff
@@ -80,49 +80,49 @@ const useGooseForm = () => {
   }, [history]);
 
   const saveForm = useCallback(async () => {
-    // TODO: Validation checks here.
-    // Send data to server:
 
     try {
-      // files added 
-      const files = images;
+      // image files added 
+      const newlyAddedImages = images;
   
-      // all files uploaded 
-      let imageUrls2 = [];
+      // all image files uploaded 
+      let newlyAddedImageUrls = [];
       
-      if (files.length > 0 || imageUrls.length > 0) {
+      if (newlyAddedImages.length > 0) {
+        
           // when no new files are added
-         if (files.length <= 0){
-          const imgArr = imageUrls.map((url) => url.imgDir)
-          imageUrls2 = imgArr
-          setImageUrls(imageUrls2)
-         }
+        //  if (newlyAddedImages.length <= 0){
+        //   const imgArr = imageUrls.map((url) => url.imgDir)
+        //   imageUrls2 = imgArr
+        //   // console.log(imageUrls)
+        //   console.log(imageUrls2)
+        //   setImageUrls(imageUrls)
+        //  }
          
-        //  when new files are added, send them to api
-        if (files.length >= 1) {
         const data = new FormData();
 
-        files.forEach((file) => {
+        newlyAddedImages.forEach((file) => {
           data.append('files', file, file.name);
         });
 
         const res = await api.formUpload(data);
 
-        imageUrls2 = res.data.data;
+        // array of added image urls
+        newlyAddedImageUrls = res.data.data;
+      
 
-        if (imageUrls2.length === 0) {
+        if (newlyAddedImageUrls.length === 0) {
           throw new Error('Failed to upload image.');
         }
       }
-      }
     
       const gooseInfo = {
         name: gooseName,
         description,
         updatedAt: Date.now(),
       };
       
-      if (gooseName && description && imageUrls2.length > 0) {
+      if (gooseName && description && (newlyAddedImageUrls.length > 0 || imageUrls.length > 0)) {
```

Comment:
A good idea might be to add a comment saying what this if statement computes (e.g., if this is a previous or new posting)